config/oauth2_config.xml.sample

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<OAuth2.0>
+    <provider name="Google">
+        <!-- Keep the client_secret file on a path inaccessible to public.  -->
+        <client_secret_file>./oauth2/client_secret.json</client_secret_file>
+        <redirect_uri>https://usegalaxy.org/oauth2callback/google</redirect_uri>
```

Comment:
This URL could be constructed, and not admin-supplied. You can use something like `web.url_for` e.g. https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/error_reports/plugins/sentry.py#L77 to construct a URL to a route. Ideally the URL could be constructed completely from the provider name (assuming provider name == python file name).